common/db/tag_store/dynamo.go

```diff
@@ -39,24 +46,33 @@ func (d *DynamoTagStore) Clear() error {
 }
 
 func (d *DynamoTagStore) Insert(tag *models.Tag) error {
+	// ensure we perform a consistent read after a write
+	defer d.setConsistentRead(true)
+
 	tag.TagID = time.Now().UnixNano()
 	return d.table.Put(tag).Run()
 }
 
 func (d *DynamoTagStore) Delete(tagID int64) error {
+	// ensure we perform a consistent read after a delete
+	defer d.setConsistentRead(true)
 	return d.table.Delete("TagID", tagID).Run()
 }
 
 func (d *DynamoTagStore) SelectAll() (models.Tags, error) {
+	defer d.setConsistentRead(false)
+
 	tags := models.Tags{}
-	if err := d.table.Scan().Consistent(true).All(&tags); err != nil {
+	if err := d.table.Scan().Consistent(d.consistentRead).All(&tags); err != nil {
 		return nil, err
 	}
 
 	return tags, nil
 }
 
 func (d *DynamoTagStore) SelectByQuery(entityType, entityID string) (models.Tags, error) {
+	defer d.setConsistentRead(false)
+
 	scan := d.table.Scan()
```

Comment:
yeah if yuo could narrow down items in this Scan by using Environment id as a partition key and querying on environment instead of scanning here, i think it would make a huge difference. 

(granted this is just based on my general understanding of the dynamo concepts of scan and query. i'm not sure how the GO api behaves here when you ask for Scan().All())